packages/api/amplify_api/test/dart_graphql_test.dart

```diff
@@ -184,12 +223,34 @@ void main() {
       expect(res.data?.id, _modelQueryId);
       expect(res.errors, equals(null));
     });
+
+    test('subscribe() should decode model data', () async {
+      Completer<void> establishedCompleter = Completer();
+      Completer<Post> dataCompleter = Completer();
+      final subscriptionRequest = ModelSubscriptions.onCreate(Post.classType);
+      final subscription = Amplify.API.subscribe(
+        subscriptionRequest,
+        onEstablished: () => establishedCompleter.complete(),
```

Comment:
This pattern of completer/complete/expectLater can be simplified with the `expectAsync*` methods from `package:test` e.g. this does the same thing.

```dart
Amplify.API.subscribe(
  subscriptionRequest,
  onEstablished: expectAsync0(() {}),
);
```

It makes more sense when the callback takes parameters and you want to run more `expect`s on those parameters, but it works here to shorten the amount of code you need to write.

Reply:
good tip, thanks, integrated this in some places but in other places it was easier to just have the completer/future.